### ISSUE TITLE
[ERROR] Error appending data to tar archive. Reason: paths in archives must not have `..` when setting path for scratch

### ISSUE THREAD

**Issue by Akkuma** [stepchowfun/toast#273]
**Description**
Trying to set the input_paths to use a directory relative to above where toast.yml lives. 

**Instructions to reproduce the bug**
A clear explanation of how to reproduce the bug.

**System information:**
 - OS: macOS Mojave 10.14.5
 - Docker version: 18.09.2
 - Toast version: latest

**Additional context**
I'm trying to have multiple toast files for multiple containers, 1 for the server, 1 for the db. I figured I'd organize the toast files into their own directory to keep things simple and for the server need to copy over the source files.

**Comment by stepchowfun** [comment]
Hi @Akkuma, thanks for the bug report! But this is actually the intended behavior—though the error message deserves improvement. The paths copied into the container mirror the corresponding paths on the host. For example, `foo/bar.txt` on the host becomes `<location>/foo/bar.txt` in the container. The problem with allowing paths to refer to ancestor directories is that you might end up with a path that goes higher than the root of the container. For example, if `location` is `/scratch`, then `../../foo.txt` on the host would become `/scratch/../../foo.txt` in the container (i.e., `/../foo.txt`), which doesn't make sense. So Toast disallows this.

I'll leave this issue open for a few days to allow you chance to respond in case I've missed something. I think the right solution in your case is just to move the toastfiles higher in the filesystem, but I'm open to discussing other ideas.

**Comment by Akkuma** [comment]
That 100% makes sense. The current solution though seems to not fair well in the scenario I presented of separate containers for differing parts of the project. Your top level repo directory would have x toast files for each part you "toast". With that in mind, obviously the current behaviour is intended, so it would make more sense to consider this more so an enhancement/feature request or open a new issue for that sort of enhancement/feature if it makes sense.

Technically toast can support the multiple containers through multiple toast files. What would be nifty is if toast could support something akin to docker compose of multiple containers. Maybe a single toast yaml that can support multiple images to iterate through or build in parallel, or a toast-compose file or something else entirely.

**Comment by stepchowfun** [comment]
@Akkuma I'll think about that for a while. In the meantime, could a simple shell script do the composition for you? I'm imagining something like this:

`run.sh` (terminate with `Ctrl+C`):

```sh
#!/usr/bin/env bash
set -euo pipefail

trap '' INT

toast -f server.yml &
SERVER_PID=$!

toast -f client.yml &
CLIENT_PID=$!

wait
```

I managed to get that to work with the following setup:

`server.yml`:

```yaml
image: nginx
tasks:
  run:
    cache: false
    input_paths:
      - index.html
    ports:
      - 3000:80
    location: /usr/share/nginx/html/
    command: nginx -g 'daemon off;'
```

`client.yml`:

```yaml
image: ubuntu
tasks:
  install_curl:
    command: |
      apt-get update
      apt-get install --yes curl

  run:
    dependencies:
      - install_curl
    command: |
      while true; do
        curl 172.17.0.1:3000
        sleep 1
      done
```

`index.html`:

```html
<!DOCTYPE html>
<html>
  <head>
    <title>Welcome to Toast!</title>
  </head>
  <body>
    <p>Hello, World!</p>
  </body>
</html>
```

**Comment by Akkuma** [comment]
Yea that definitely works for now, thanks for the help!

**Comment by stepchowfun** [comment]
Great, then I'll close the issue and improve the error message you encountered.

Just wondering, is there anything not ideal for you about using a simple shell script like that to run two instances of Toast? I'm trying to assess what benefit there would be to building this into Toast itself.
